pkg/userpreferences/userpreferences.go

```diff
@@ -0,0 +1,29 @@
+package userpreferences
+
+import (
+	"net/http"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/steve/pkg/stores/proxy"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+type UserPreference struct {
+	Data map[string]string `json:"data"`
+}
+
+func Register(schemas *types.APISchemas, cg proxy.ClientGetter) {
+	schemas.InternalSchemas.TypeName("userpreference", UserPreference{})
```

Comment:
Why do we need this for bumping the dependency?

Reply:
this has been removed from the steve API to avoid duplication in rancher, but the `userpreference` API is required. 
https://github.com/rancher/steve/commit/e106b9e16bf59d8b128b2de038f81ec7f6671bcb